From dc60b6d56fc8cbb9f74e5489da28ac86aa8e9912 Mon Sep 17 00:00:00 2001
From: T Karthik Reddy <t.karthik.reddy@xilinx.com>
Date: Mon, 23 Mar 2020 02:22:15 -0600
Subject: [PATCH] arm64: zynq_sdhci: Resolve failed mmc tuning due to gcc-9.2

Assigning set of default tapdelays with macros causing sd/emmc tuning
failed due to garbage values assigned for tuning at runtime with gcc-9.2
compiler. So assign default tapdelays to array of unsigned integers instead
of assigning them with macros.

Signed-off-by: T Karthik Reddy <t.karthik.reddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/mmc/zynq_sdhci.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/drivers/mmc/zynq_sdhci.c b/drivers/mmc/zynq_sdhci.c
index 2293964b82..9f3f375841 100644
--- a/drivers/mmc/zynq_sdhci.c
+++ b/drivers/mmc/zynq_sdhci.c
@@ -17,14 +17,14 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
-#define ZYNQMP_ITAP_DELAYS		{0x0, 0x15, 0x0, 0x0, 0x3D, 0x0,\
-					 0x15, 0x12, 0x15}
-#define ZYNQMP_OTAP_DELAYS		{0x0, 0x5, 0x3, 0x3, 0x4, 0x3,\
-					 0x5, 0x6, 0x6}
-#define VERSAL_ITAP_DELAYS		{0x0, 0x2C, 0x0, 0x0, 0x36, 0x0,\
-					 0x2C, 0x1E, 0x2C}
-#define VERSAL_OTAP_DELAYS		{0x0, 0x4, 0x3, 0x2, 0x3, 0x2,\
-					 0x4, 0x5, 0x5}
+static u32 zynqmp_itap_delays[] = { 0x0, 0x15, 0x0, 0x0, 0x3D, 0x0, 0x15, 0x12,
+				    0x15 };
+static u32 zynqmp_otap_delays[] = { 0x0, 0x5, 0x3, 0x3, 0x4, 0x3, 0x5, 0x6, 0x6
+				  };
+static u32 versal_itap_delays[] = { 0x0, 0x2C, 0x0, 0x0, 0x36, 0x0, 0x2C, 0x1E,
+				    0x2C};
+static u32 versal_otap_delays[] = { 0x0, 0x4, 0x3, 0x2, 0x3, 0x2, 0x4, 0x5, 0x5
+				  };
 
 #define MMC_BANK2			0x2
 
@@ -269,11 +269,11 @@ static void arasan_zynqmp_dt_parse_tap_delays(struct udevice *dev)
 	int i;
 
 	if (ofnode_device_is_compatible(dev_ofnode(dev), "xlnx,zynqmp-8.9a")) {
-		itapdly = (u32 [MMC_MAX_BUS_SPEED]) ZYNQMP_ITAP_DELAYS;
-		otapdly = (u32 [MMC_MAX_BUS_SPEED]) ZYNQMP_OTAP_DELAYS;
+		itapdly = zynqmp_itap_delays;
+		otapdly = zynqmp_otap_delays;
 	} else {
-		itapdly = (u32 [MMC_MAX_BUS_SPEED]) VERSAL_ITAP_DELAYS;
-		otapdly = (u32 [MMC_MAX_BUS_SPEED]) VERSAL_OTAP_DELAYS;
+		itapdly = versal_itap_delays;
+		otapdly = versal_otap_delays;
 	}
 
 	/* as of now bank2 tap delays are same for zynqmp and versal */
-- 
2.17.1

